### PR TITLE
DP-28044: Make it possible for the chatbot to show up on Collections

### PR DESCRIPTION
**Pull request by arthurbaghdas**
**Description:**
Explain the technical implementation of the work done.

**Jira:** (Skip unless you are MA staff)
DP-28044

**To Test:**
- [ ] Add steps to test this feature

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

**Comment by tom-fleming** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Add a changelog YAML file to this PR</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d00df6ddd929d1dc8a95f78b9d7ee37c93e94cdc
</p>

File: docroot/modules/custom/mass_content/mass_content.module [hunk 1]
```diff
@@ -639,7 +639,7 @@ function mass_content_form_taxonomy_term_collections_form_alter(array &$form, Fo
     $form['field_sort_order']['#access'] = FALSE;
     $form['field_primary_parent']['#access'] = FALSE;
     $form['field_landing_page']['#access'] = FALSE;
-    $form['field_collection_logo']['#access'] = FALSE;
```
Comment (arthurbaghdas) [inline docroot/modules/custom/mass_content/mass_content.module:642]: There is a duplicate line below, that's why I've removed this.

File: docroot/modules/custom/mass_metatag/tests/src/ExistingSite/CollectionsMetadataTest.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ * Collections metadata tests.
+ */
+class CollectionsMetadataTest extends MetadataTestCase {
+
+  use TestContentTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getContent(): ContentEntityInterface {
```
Comment (weitzman) [inline docroot/modules/custom/mass_metatag/tests/src/ExistingSite/CollectionsMetadataTest.php:20]:
```suggestion
  public function createContent(): ContentEntityInterface {
```

File: docroot/modules/custom/mass_metatag/tests/src/ExistingSite/CollectionsMetadataTest.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    ]);
+
+    return $collection_term;
+  }
+
+  /**
+   * Asserts that a collections page produces the metadata we expect.
+   */
+  public function testHasExpectedMetadata() {
+    $entity = $this->getContent();
```
Comment (weitzman) [inline docroot/modules/custom/mass_metatag/tests/src/ExistingSite/CollectionsMetadataTest.php:37]:
```suggestion
    $entity = $this->createContent();
```

File: docroot/themes/custom/mass_theme/mass_theme.theme [hunk 4]
```diff
@@ -5982,6 +5982,19 @@ function mass_theme_preprocess_views_view_collection_all_page_all(&$variables) {
   $tid = reset($res);
   $term = Term::load($tid);
 
+  // Attach mg_labels metatag, if it's available.
```
Comment (weitzman) [inline docroot/themes/custom/mass_theme/mass_theme.theme]: I'm a little surprised to see token replacement in mass_theme_preprocess_views_view_collection_all_page_all. Doesn't metatag have a more standard way that uses config?

Comment (arthurbaghdas) [reply docroot/themes/custom/mass_theme/mass_theme.theme]: Removed this part and enabled views display extender of the metatag_views module. Configured it that way.

File: docroot/modules/custom/mass_metatag/tests/src/ExistingSite/CollectionsMetadataTest.php (hunk 2 above)
Comment (arthurbaghdas) [reply docroot/modules/custom/mass_metatag/tests/src/ExistingSite/CollectionsMetadataTest.php:20]: Renaming this causes issues, we should go deeper and rename the base method and everywhere it is used. So reverting to what it was

File: docroot/modules/custom/mass_metatag/tests/src/ExistingSite/CollectionsMetadataTest.php (hunk 3 above)
Comment (arthurbaghdas) [reply docroot/modules/custom/mass_metatag/tests/src/ExistingSite/CollectionsMetadataTest.php:37]: Renaming this causes issues, we should go deeper and rename the base method and everywhere it is used. So reverting to what it was

File: docroot/modules/custom/mass_content/mass_content.module [hunk 5]
```diff
@@ -639,7 +639,7 @@ function mass_content_form_taxonomy_term_collections_form_alter(array &$form, Fo
...
+    $form['field_reusable_label']['#access'] = FALSE;
```
Comment (weitzman) [inline docroot/modules/custom/mass_content/mass_content.module:642]: This is fine for now. Next time I think its better to remove from the form display.

File: docroot/modules/custom/mass_metatag/mass_metatag.module [hunk 6]
```diff
@@ -81,6 +88,46 @@ function mass_metatag_token_info() {
...
+            if ($data['term'] instanceof ContentEntityInterface) {
+              $term = $data['term'];
+            }
+          }
+          elseif ($collection = \Drupal::routeMatch()->getParameter('collection')) {
+            // Get collection terms matching the field_url_name
+            $query = Drupal::entityQuery('taxonomy_term');
+            $query
+              ->accessCheck(FALSE)
+              ->condition('vid', 'collections')
```
Comment (weitzman) [inline docroot/modules/custom/mass_metatag/mass_metatag.module:106]: Do we want to restrict to just published terms?

**Review by weitzman — APPROVED** [review]
Asked a minor question. This is approved merge regardless.

File: docroot/modules/custom/mass_content/mass_content.module (hunk 5 above)
Comment (arthurbaghdas) [reply docroot/modules/custom/mass_content/mass_content.module:642]: I think the code conditionally removes the fields, which is hard to achieve from the UI

File: docroot/modules/custom/mass_metatag/mass_metatag.module (hunk 6 above)
Comment (arthurbaghdas) [reply docroot/modules/custom/mass_metatag/mass_metatag.module:106]: This actually duplicates the code from another part, so we can move forward and come back with a separate ticket if we need to. If no objections I'll merge

**Pull request merged**